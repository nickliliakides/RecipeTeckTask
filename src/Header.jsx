import React, { useState } from 'react';
import './Header.styles.scss';

const Header = ({ recipes, allRecipes, setRecipes }) => {
  const [searchField, setSearchField] = useState('');

  const filterRecipes = (recipes, e) => {
    const filteredRecipes = recipes.filter((recipe) => {
      const matchIngredient = recipe.ingredients.some((ing) =>
        ing.toLowerCase().includes(e.currentTarget.value.trim().toLowerCase())
      );

      const matchTitle = recipe.tags.some((tag) =>
        tag.toLowerCase().includes(e.currentTarget.value.trim().toLowerCase())
      );

      return (
        recipe.title
          .toLowerCase()
          .includes(e.currentTarget.value.trim().toLowerCase()) ||
        matchIngredient ||
        matchTitle
      );
    });
    if (filteredRecipes.length) setRecipes(filteredRecipes);
  };

  const handleChange = (e) => {
    setSearchField(e.currentTarget.value);
    if (searchField.length < e.currentTarget.value.length)
      filterRecipes(recipes, e);
    else filterRecipes(allRecipes, e);
  };

  return (
    <header className='header'>
      <img src='img/recipeLogo.png' alt='Logo' className='header__logo' />
      <div className='search'>
        <input
          type='text'
          className='search__field'
          placeholder='Search over recipes...'
          onChange={(e) => handleChange(e)}
        />
        <div className='search__icon__wrapper'>
          <svg className='search__icon'>
            <use href='img/icons.svg#icon-magnifying-glass'></use>
          </svg>
        </div>
      </div>
    </header>
  );
};

export default Header;
