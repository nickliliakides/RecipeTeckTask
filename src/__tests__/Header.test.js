import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';
import recipes from '../mockRecipes';

let wrapper;

const defaultProps = {
  recipes,
  allRecipes: recipes,
  setRecipes: jest.fn(),
};

describe('<Header /> component', () => {
  beforeEach(() => {
    wrapper = shallow(<Header {...defaultProps} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should matches its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
