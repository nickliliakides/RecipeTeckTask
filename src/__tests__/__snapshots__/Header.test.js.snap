// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component should matches its snapshot 1`] = `
<header
  className="header"
>
  <img
    alt="Logo"
    className="header__logo"
    src="img/recipeLogo.png"
  />
  <div
    className="search"
  >
    <input
      className="search__field"
      onChange={[Function]}
      placeholder="Search over recipes..."
      type="text"
    />
    <div
      className="search__icon__wrapper"
    >
      <svg
        className="search__icon"
      >
        <use
          href="img/icons.svg#icon-magnifying-glass"
        />
      </svg>
    </div>
  </div>
</header>
`;
